File: backend/src/main/java/com/codecool/umbrella/api/endpoint/AdminEndpoint.java
package com.codecool.umbrella.api.endpoint;

import com.codecool.umbrella.logic.AdminService;
import com.codecool.umbrella.model.User;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/admin")
@RequiredArgsConstructor
public class AdminEndpoint {

    private final AdminService adminService;

    @GetMapping
    public List<User> getAllUsers() {
        return adminService.getAllUsers();
    }

    @DeleteMapping("/{username}/delete")
    public void deleteUser(@PathVariable("username") String username) {
        adminService.removeUser(username);
    }

    @PutMapping("/{username}/role/{newRole}")
    public void changeUserRole(@PathVariable("username") String username,
                               @PathVariable("newRole") String newRole) {
        adminService.changeUserRole(username, newRole);
    }

}

File: frontend/src/fetch/adminEndpoint.js
import {axiosInstance, url} from "./axiosInstance"

export function getAllUsers() {
  return axiosInstance.get(`${url}/api/admin`)
    .then(response => {
      return response.data;
    })
    .catch(error => {
      console.log("Error: " + error);
      return [];
    });
}

export function changeUserRole(username, newRole) {
  return axiosInstance.put(`${url}/api/admin/${username}/role/${newRole}`)
    .then(response => {
      return response.status;
    })
    .catch(error => {
      console.log("Error: " + error);
      return error.response.status;
    });
}

export function deleteUser(username) {
  return axiosInstance.delete(`${url}/api/admin/${username}/delete`)
    .then(response => {
      return response.status;
    })
    .catch(error => {
      console.log("Error: " + error);
      return error.response.status;
    })
}
